src/main/kotlin/no/fdk/fdk_event_harvester/adapter/EventAdapter.kt

```diff
@@ -0,0 +1,36 @@
+package no.fdk.fdk_event_harvester.adapter
+
+import no.fdk.fdk_event_harvester.model.HarvestDataSource
+import org.slf4j.LoggerFactory
+import org.springframework.http.HttpStatus
+import org.springframework.stereotype.Service
+import java.io.BufferedReader
+import java.net.HttpURLConnection
+import java.net.URL
+
+private val LOGGER = LoggerFactory.getLogger(EventAdapter::class.java)
+
+@Service
+class EventAdapter {
+
+    fun getEvents(source: HarvestDataSource): String? =
+        try {
+            val connection = URL(source.url).openConnection() as HttpURLConnection
```

Comment:
Sjekk ut try-with-resources: https://www.baeldung.com/java-try-with-resources

Muligens finnes det en enda finere måte i Kotlin

Reply:
Vurder også å bruke connection pool: https://www.baeldung.com/httpclient-connection-management

Reply:
try-with-resources er ikke en greie i kotlin, `use` som blir brukt på inputStream lengre ned er det eventuelle alternativet for å automatisk lukke ressurser. Det er dog ikke mulig for HttpUrlConnction, er en annen type closeable enn hva kotlin støtter.

Men burde vel uansett legge inn en `disconnect()`